packages/@aws-cdk/aws-events-targets/test/stepfunctions/statemachine.test.ts

```diff
@@ -28,3 +29,32 @@ test('State machine can be used as Event Rule target', () => {
     ],
   });
 });
+
+test('Existing role can be used for State machine Rule target', () => {
+  // GIVEN
+  const stack = new cdk.Stack();
+  const rule = new events.Rule(stack, 'Rule', {
+    schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+  });
+  const role = new iam.Role(stack, 'Role', {
+    assumedBy: new iam.ServicePrincipal('events.amazonaws.com'),
+  });
+  const stateMachine = new sfn.StateMachine(stack, 'SM', {
+    definition: new sfn.Wait(stack, 'Hello', { time: sfn.WaitTime.duration(cdk.Duration.seconds(10)) }),
+    role,
+  });
+
+  // WHEN
+  rule.addTarget(new targets.SfnStateMachine(stateMachine, {
+    input: events.RuleTargetInput.fromObject({ SomeParam: 'SomeValue' }),
+  }));
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Events::Rule', {
+    Targets: [
+      {
+        Input: '{"SomeParam":"SomeValue"}',
+      },
+    ],
+  });
```

Comment:
`grantStartExecution` should result in ad added `AWS::IAM::Policy` with the permissions to `startExecution`, scoped to the state machine arn, and mapped to the provided role. this test should be verifying that mapping as well.

Reply:
Done